### PR TITLE
tio: submission

### PR DESCRIPTION
**Pull request by rhapsodyv**
#### Description

New port submission for https://github.com/tio/tio

###### Type(s)

- [x] submission

###### Tested on

macOS 12.4 21F79 arm64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: comms/tio/Portfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           meson 1.0
+
+name                tio
+version             1.45
+revision            1
```
Comment (herbygillot) [inline comms/tio/Portfile:8]: Since this is a brand new port, `revision` should be starting at 0.

File: comms/tio/Portfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use_xz              yes
+
+description         Simple TTY terminal I/O application
+long_description    tio is a simple serial device tool which \
+                    features a straightforward command-line \
+                    and configuration file interface to easily \
+                    connect to serial TTY devices for basic \
+                    I/O operations.
+
+checksums           sha256  4e72a946792346f86504865ed2923688f4dd9281905bb81e76ba3fd6f3b96014 \
```
Comment (herbygillot) [inline comms/tio/Portfile:25]: You also need to add the `rmd160` checksum.  You can use `port bump tio` to automatically fill in the correct checksum, or you can calculate it manually using `openssl`: `openssl dgst -rmd160 <file>`

**Comment by rhapsodyv** [comment]
@herbygillot thanks for your review!

**Comment by herbygillot** [comment]
Hey @rhapsodyv, this is looking ready to merge - there was one more thing I forgot to mention, and then we're good.  Thank you.

**Comment by ryandesign** [comment]
> Doing this allows you to remove the name, version, master_sites, distfiles & use_xz fields as github.setup will automatically populate those implicitly.

@herbygillot The github portgroup will never set `use_xz` automatically. Its defaults are to use an automatically-generated tar.gz distfile from GitHub's "tarball" endpoint, which is deprecated. Ports that must use an automatically-generated tarball should request the "archive" endpoint instead of the "tarball" endpoint by setting `github.tarball_from archive`. This will change the checksums and would need to be handled as a [stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates) if the filename is not changed at the same time, for example by updating the version.

However the developers of tio have offered a [manually-created tar.xz release download](https://github.com/tio/tio/releases/tag/v1.46) (look in the Assets section), and release downloads should always be preferred to automatically-generated tarballs when available. To do this, set `github.tarball_from releases` and `use_xz yes`. This will change the checksums but since the filename also changes it does not need to be treated as a stealth update.

**Comment by rhapsodyv** [comment]
@ryandesign thanks for your feedback! I will fix the port and add the lib dependency.

**Pull request merged**